### DISCUSSION TITLE
사용자 위치 정보 관련

### DISCUSSION THREAD

**Discussion by psh2849** [euphony-io/check-in-speaker#12] [General]
멘토님이 말씀해주신 아이디어를 토대로 어떻게 구현하면 좋을지 생각해보았습니다!

**1. 체크인 버튼 클릭 시**
체크인 버튼 클릭 시 `FusedLocationProviderClient` 객체를 이용해서 
사용자의 위치 정보(longitude, latitude)를 획득할 수 있습니다.

이 위치 정보를 https://developers.kakao.com/docs/latest/ko/local/dev-guide#coord-to-address에서 `좌표로 주소 변환하기` 부분을 통해서 좌표를 주소로 변환할 수 있을 것 같습니다.

변환한 주소 값과, 체크인 버튼을 누른 시간을 함께 `RecyclerView`에 저장하면 구현이 가능하지 않을까..... 생각해봤습니다

고려할 점은 kakao api에서 받은 api key를 잘 관리할 수 있는 방법이 필요할 것 같습니다 ㅎ

**Comment by jopopcorn** [comment]
오호 카카오에서 API를 제공하는군요! 저는 Geocoder를 통해서 주소로 변환하려고 했는데 API를 사용해도 괜찮을 것 같네요!

**Reply by psh2849** [reply]
Geocoder는 몰랐는데 같은 기능을 하는 api가 있었네요
주소 값만 얻는 거라면 이것도 괜찮을 것 같습니다!

**Reply by jopopcorn** [reply]
넵넵 API를 이용하지 않아도 내부 라이브러리에서 이미 제공하고 있는 기능이어서 Geocoder로 해도 될 것 같습니다!

**Comment by jopopcorn** [comment]
API 키 관리 얘기가 나와서 추가로 코멘트 달아봅니다!
저는 다른 프로젝트에서 API 킷값을 local.properties에 저장했는데 이 방법이 가장 간단하더라고요..ㅎㅎ
다른 간단한 방법으로는 base64나 SHA256 같은 알고리즘으로 암호화 하는 방법이고, 제대로 처리하려면 ndk쪽을 알아야 한다고 들었습니다!
그리고 제가 들은 바로는 실무에서는 버전을 올릴 때마다 API 킷값을 갱신한다고 합니다! 킷값 관리도 중요하기 때문에 알아두면 좋을 거 같아요 😸

**Reply by psh2849** [reply]
오.. 좋은 정보 감사합니다 👍👍👍👍👍

**Comment by jopopcorn** [comment]
@psh2849 
사용자 위치 정보 관련한 기능 목록은 다음과 같습니다!
이 중에서 Room과 가장 관련된 2, 4번째 기능 개발을 맡아주실 수 있을까요?
이제서야 시간이 났는데 일요일까지 마무리짓는데 시간이 부족할 것 같아서 뒤늦게 역할을 더 세분화하고자 합니다 ㅜ
역할 분배 얘기 먼저 말씀주셔서 감사드려요!!

**위치 정보 수집 기능**
- 앱을 처음 실행했을 때 GPS 권한 받기
- Room을 사용해 방문한 위치와 날짜를 저장하는 데이터 설계 + 저장 및 불러오기 쿼리 작성
- 체크인 시작일 때(음파를 생성 중일 때) 방문 위치를 장소로 변환하기 & 날짜 문자열 생성
- 체크인 시작일 때(음파를 생성 중일 때) `방문 위치를 장소로 변환한 문자열`과 `날짜` Room에 저장하기
- `VisitRecordActivity` 화면의 리싸이클러뷰에 저장 중인 데이터를 출력하기 (Room의 데이터 불러오기)

**Reply by psh2849** [reply]
넵! 그래도 저는 많이 배운다는 생각으로 하고 있습니다 ㅎㅎ..
ViewModel에 Room을 적용시키는 것도 처음이라 간간히 모르는 부분이 있으면 좀 여쭤보도록 할게요!
2, 4번째 기능은 최대한 빨리 끝내보도록 하겠습니다!

**Reply by jopopcorn** [reply]
저도 룸은 사용해본 적이 없어서 모르는 게 많지만 답변드릴 수 있는 부분이라면 최대한 답 드리겠습니다!
무리하지 마시고 저도 최대한 해볼 수 있는 만큼 개발하고 정규 회의 들어가겠습니다!
